### ISSUE TITLE
Updating Continuous Deployment

### ISSUE THREAD

**Issue by Heliotrop3** [beartype/beartype#22]
Both are taken from beartype\\.github\workflows\pythonrelease.yml
```
on:
  push:
    #FIXME: Uncomment after determining why this induces critical failures on
    #"git push --tags" attempts with remote error messages resembling:
    #    error: dst refspec refs/heads/master matches more than one
    #    error: failed to push some refs to 'git@github.com:beartype/beartype.git'
    #See also: https://github.com/actions/create-release/issues/13
    # branches: [ master ]

    # Sequence of glob expressions matched against "refs/tags" pushed to the
    # branches above.
    tags:
      - 'v*'   # Match "v"-prefixed tags (e.g., "v6.9.6").
```

```
      #FIXME: Enable after integrating this "pypi" job into the "release" job
      #above. In theory, this should be trivial. Let's test this later, eh?
      # - name: "Publishing GitHub release assets from tag..."
      #   uses: actions/upload-release-asset@v1.0.1
      #   env:
      #     GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
      #   with:
      #     upload_url: ${{ steps.create_release.outputs.upload_url }} # This pulls from the CREATE RELEASE step above, 
                                                                       # referencing it's ID to get its outputs object, 
                                                                       # which include an `upload_url`. 
                                                                       # See this blog post for more info: https://jasonet.co/posts/new-features-of-github-actions/#passing-data-to-future-steps
      #     asset_path: "dist/beartype-${{ github.ref }}.zip"
      #     asset_name: "beartype-${{ github.ref }}.zip"
      #     asset_content_type: application/zip
```
[An update from the repository](https://github.com/actions/create-release/issues/119#issue-797229040)  of [the referenced](https://github.com/actions/create-release/issues/13) issue notes it is no longer actively maintained and "In the last year or two since we originally published this, a large number of really good community Actions have appeared on the Marketplace for doing this." 

Thus, resolution of creating a release via Github actions will almost certainly require researching alternative solutions on the Github Marketplace.

Additional Links:
[Windows Package Manager (Chocolatey)](https://github.com/marketplace/actions/chocolatey-action)
[PyPi Deployment](https://github.com/marketplace/actions/pypi-deployment)
[Github Deployments](https://github.com/marketplace/actions/github-deployments)
[Semaphore](https://github.com/marketplace/semaphore)

See also #20 covering improvements to CI

**Comment by Heliotrop3** [comment]
@leycec, not very familiar with the CI/CD so more getting this out there as a contact point for someone who does. In short, I'm not certain how the repository for creating a release going EOL effects our CI/CD setup.

**Comment by leycec** [comment]
*Yikes!* Thanks a metric-standardized bundle for catching this, Tyler. You remain the hero we both deserved and gratefully received. As you already suspect, `create-release` going EOL affects our CD setup by guaranteeing **that setup will catastrophically blow up in my face during some future version bump.**

We now need to deeply refactor that setup to leverage an actively maintained release action *before* releasing `beartype 0.5.2`. This is actually fine for certain loose definitions of "fine," because `create-release` didn't support formatting release messages as Markdown, which meant I had to manually do that on every bump, which *was so annoying it made me want to chew my own face off.*

Thankfully, I didn't do that. Let's casually grep the [GitHub Marketplace](https://github.com/marketplace) and see what immediately leaps out and tries to chew our faces off.

**Comment by leycec** [comment]
`</wipes_sweaty_brow>`

On the bright side, [I found **Release Action** and then published a clickbait doctoral thesis about my findings](https://github.com/actions/create-release/issues/119#issuecomment-783010321). On the dark side, there goes Sunday evening, which my unplayed PS4 now wants back.

tl;dr code snippet or it didn't happen, so:

```
name: release

on:
  push:
    tags:
      - 'v*'

jobs:
  release:
    name: "Create tagged release on GitHub"
    runs-on: ubuntu-latest

    steps:
      - name: "Checking out repository..."
        uses: 'actions/checkout@v2'
      - name: "Publishing GitHub release from tag..."
        uses: 'actions/release-action@v1'
        with:
          name: "beartype ${{ github.ref }}"
          body: ${{ github.event.commits[0].message }}
          token: ${{ secrets.GITHUB_TOKEN }}
```

What could possibly go wrong, Murphy? :grimacing:

**Comment by leycec** [comment]
I'd intended to delay this migration until *after* publishing our next stable release, because I am intrinsically lazy. But the author of `create-release` effectively just archived that repository, which inspires justifiable fear rather than unshakeable confidence.

We're immediately migrating to @ncipollo's [`release-action`](https://github.com/ncipollo/release-action), which appears to be the sanest, most configurable, *and* most actively-maintained release action. Thanks, @ncipollo! Have all my grateful repository stars. :star2:

**Comment by leycec** [comment]
**Resolved by 3dd551e.** Thanks for all the quality assurance, @Heliotrop3! You keep us honest, even when we'd rather not be. :smiling_face_with_three_hearts:
